testing/jormungandr-integration-tests/src/common/jormungandr/process.rs

```diff
@@ -239,21 +240,55 @@ impl JormungandrProcess {
 
 impl Drop for JormungandrProcess {
     fn drop(&mut self) {
-        let errors = self
-            .logger
-            .get_lines_with_level(LogLevel::ERROR)
-            .collect::<Vec<_>>();
-        if !errors.is_empty() {
-            println!("Error lines:");
-            for line in errors {
-                println!("{}", line);
-            }
-        }
+        use rand::Rng;
         // There's no kill like overkill
         let _ = self.child.kill();
-
         // FIXME: These should be better done in a test harness
         self.child.wait().unwrap();
+        if panicking() {
+            let dir_name = rand::thread_rng()
```

Comment:
Just use `tempfile`, we already use it across all of our projects. It will take care of generating a name and creating a directory. Hint: [`TempDir::into_path`](https://docs.rs/tempfile/3.2.0/tempfile/struct.TempDir.html#method.into_path) prevents a temporary directory from being removed when a `TempDir` goes out of scope.